### DISCUSSION TITLE
What if you await for a "job?"

### DISCUSSION THREAD

**Discussion by TitanDvd** [livewire/livewire#6306] [3. Feature ideas]
I came up with this idea to deal in some way with async tasks. I had the need for the user to request a download file but I wanted to give the  possibility that user hadn't to stay on the same page BEFORE download starts. For sure, the user can leave the page after downloading starts but no before or else file won't ever get downloaded... so

An idea could be
1. When user creates in its component a `LwJobs->do(function(){})` or  `LwJobs->do([function(){}, function(){}])`  then Lw could store at some place, like local storage, a pice of information that indicates how many "jobs" it have to watch for completation, the order of completations if needed and other stuff.
2. Lw must check in every windows load if there is any "job" to check if completed and then return the previous stored result if any
3. Do a poll for stored "jobs" besides every windows-load check
4. The "job" results can reside on filesystem for convenience or database
5. The stored results will probably be php instructions or files ready to download

I'm aware of many libraries that do this, but I wish Lw could have it natively.

**Comment by wizzymore** [comment]
You can do this using Laravel jobs, files or database for storage (an example: generating a key for the specific task, creating an empty file in a temp folder named as the key you just generated and deleting the file in your job when the action is done. If you need output from the job i recommend the opposite, poll the component while the file doesn't exist and then read the output), then you can use livewire to poll conditionally while the task is not "done"
